indico/htdocs/js/indico/jquery/itempicker.js

```diff
@@ -0,0 +1,162 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    $.widget('indico.itempicker', {
+
+        options: {
+            onSelect: null, // function called every time an item gets selected from the list
+            createItem: null, // function returning a jQuery object representing each item in the dropdown
+            items: null, // all the options that should be included in the dropdown
+            createCustomFooterElements: null,
```

Comment:
Couldn't we move most of the complexity of footer elements (like [here](https://github.com/indico/indico/pull/2198/files#diff-3211d7cc81c90c28a7cf31797a9afcddR158)) into the widget? I think we want the "footer" area not to be extremely fancy, so simply passing a list of pairs (text, callback_on_click) should be enough here.